pkg/user/tx_client.go

```diff
@@ -251,18 +289,60 @@ func SetupTxClient(
 		return nil, fmt.Errorf("failed to create signer: %w", err)
 	}
 
-	return NewTxClient(encCfg.Codec, signer, conn, encCfg.InterfaceRegistry, options...)
+	txClient, err := NewTxClient(encCfg.Codec, signer, conn, encCfg.InterfaceRegistry, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Always start the tx queue
+	if txClient.txQueue != nil {
+		if err := txClient.txQueue.start(ctx); err != nil {
+			return nil, fmt.Errorf("failed to start tx queue: %w", err)
+		}
+	}
+
+	return txClient, nil
 }
 
 // SubmitPayForBlob forms a transaction from the provided blobs, signs it, and submits it to the chain.
 // TxOptions may be provided to set the fee and gas limit.
+// This method uses the tx queue infrastructure and blocks until the transaction is confirmed.
 func (client *TxClient) SubmitPayForBlob(ctx context.Context, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error) {
-	resp, err := client.BroadcastPayForBlob(ctx, blobs, opts...)
-	if err != nil {
-		return nil, err
+	if client.txQueue == nil {
+		return nil, errors.New("tx queue not configured")
```

Comment:
NOTE: per request, there is no long a SubmitPFBParallel method, now there is only a SubmitPFB method. We are forcing users to use the queue when submitting blobs via this method.

In order to get around this, they need to call BroadcastPayForBlobWithAccount.